src/query/storage/m3/dynamic_cluster_test.go

```diff
@@ -87,13 +96,13 @@ func TestDynamicClustersInitialization(t *testing.T) {
 		SetDynamicClusterNamespaceConfiguration([]DynamicClusterNamespaceConfiguration{cfg}).
 		SetInstrumentOptions(instrument.NewOptions())
 
-	clusters, err := newDynamicClusters(opts)
+	updateCh := make(chan bool, 1)
+	clusters, err := newDynamicClusters(opts, updateCh)
 	require.NoError(t, err)
 
-	defer func() {
-		<-nsInitializer.updateCh
-		clusters.Close()
-	}()
+	defer clusters.Close()
+
+	<-updateCh
```

Comment:
Worth trying to use `xclock.Until` here instead of relying on passing in a test channel?

Reply:
Ahh, yeah, I can do that.